satpy/resample.py

```diff
@@ -823,11 +828,135 @@ def compute(self, data, expand=True, **kwargs):
                             coords=coords or None)
 
 
+class BucketResampler(BaseResampler):
+    """Base class for bucket resampling which implements averaging.
+
+    Bucket resampling calculates the average of all the values that
+    are closest to each bin and inside the target area.
+
+    Parameters
+    ----------
+    fill_value : float (default: np.nan)
+        Fill value for missing data
+    mask_all_nans : boolean (default: False)
+        Mask all locations with all-NaN values
+    """
+
+    def __init__(self, source_geo_def, target_geo_def):
+        super(BucketResampler, self).__init__(source_geo_def, target_geo_def)
+        self.resampler = None
+
+    def precompute(self, **kwargs):
+        """Create X and Y indices and store them for later use."""
+        LOG.debug("Initializing bucket resampler.")
+        source_lons, source_lats = self.source_geo_def.get_lonlats()
+        self.resampler = bucket.BucketResampler(self.target_geo_def,
+                                                source_lons,
+                                                source_lats)
+
+    def compute(self, data, fill_value=np.nan, mask_all_nan=False, **kwargs):
```

Comment:
Shouldn't this `compute` method be moved to a separate class `BucketAvgResampler(BucketResampler)`?

Reply:
Moved the `compute()` method to `BucketAvg` subclass and renamed the base class to `BucketResamplerBase`.